pkg/session/session.go

```diff
@@ -4027,6 +4027,10 @@ func GetStartTSFromSession(se any) (startTS, processInfoID uint64) {
 	if txnInfo != nil {
 		startTS = txnInfo.StartTS
 		processInfoID = txnInfo.ConnectionID
+	} else {
+		tmp.sessionVars.TxnCtxMu.Lock()
```

Comment:
`sessionVars.TxnCtx` will not be cleaned before executing another statement, so using the `StartTS` from `TxnCtx` will block the GC for very long time if some internal sessions are not used (e.g. stored in a pool).

It's possible to clean the `sessionVars.TxnCtx`. But IMO, it'd be simpler to use:

```go
txn := tmp.Txn(false)
if txn != nil {
    startTS = txn.StartTS()
}
```

It should have the same value with `txnInfo.StartTS` (ref `changePendingToValid`, it sets the `txnInfo.StartTS` with `t.StartTS()`), and it doesn't depend on the `ProcessInfo`.

Reply:
https://github.com/pingcap/tidb/pull/57044/commits/2cfe63653e9a67e221d560b7345a3cd1657617f9

Reply:
Maybe we still need `TxnInfo()` because `processInfoID` will be used to filter analyze SQL.